### PR TITLE
[FLOC-4180] Jenkins runs cloudformation tests

### PR DESCRIPTION
**Pull request by wallrj**
Fixes: https://clusterhq.atlassian.net/browse/FLOC-4180

Changes:
 * Changed the test to require only an ACCEPTANCE_YAML environment variable (credentials and region are now read from there)
 * Moved test_installer and fixtures to admin.tests 
   * (to avoid them being run along with the cluster based acceptance tests)
 * added a new cron job to build.yaml 
   * couldn't find a way to re-use the existing ``run_trial_cli`` command list
   * considered modifying jobs.groovy with a new function to insert environment variables before running the cli script...but it seemed like a rabbit hole.

The new job appears here:
 * http://ci-live.clusterhq.com:8080/job/ClusterHQ-flocker/job/jenkins-runs-cloudformation-tests-FLOC-4180/view/cron/job/_run_admin_test_test_installer/

I started it manually.

NB The acceptance.yaml parsing functions were copied from https://github.com/ClusterHQ/bookshelf/pull/8

### CODE REVIEW

File: admin/test/test_installer.py [hunk 1]
```diff
@@ -18,11 +18,12 @@
 
 from eliot import Message
 
-from ...common.runner import run_ssh
-from ...common import gather_deferreds, loop_until, retry_failure
-from ...testtools import AsyncTestCase, async_runner, random_name
-from ..testtools import connected_cluster
-
+from flocker.common.runner import run_ssh
```
Comment (myechuri) [inline admin/test/test_installer.py]: Reorg to admin/test looks great! Thanks for making the change!

File: admin/test/test_installer.py [hunk 2]
```diff
@@ -106,31 +109,54 @@ def remote_postgres(client_ip, host, command):
     )
 
 
-def get_stack_report(stack_id):
+def aws_output(args, aws_config):
```
Comment (myechuri) [inline admin/test/test_installer.py]: Nice refactoring.

File: admin/test/test_installer.py [hunk 3]
```diff
@@ -245,37 +279,49 @@ def _new_stack(self):
         environment variable. AWS credentials and CloudFormation parameter
         values must also be supplied as environment variables.
```
Comment (myechuri) [inline admin/test/test_installer.py]: Can you please consider updating this comment to reflect the fact that AWS credentials are now gathered from acceptance.yml? Thanks!

File: build.yaml [hunk 4]
```diff
@@ -1600,6 +1600,32 @@ job_type:
...
+                   'export ACCEPTANCE_YAML=/tmp/acceptance.yml',
+                   *run_trial_with_coverage,
+                   *run_coverage,
+                   *convert_results_to_junit ]
+        }
+      clean_repo: true
+      archive_artifacts: *flocker_artifacts
+      publish_test_results: true
+      # Reasoning as for run_acceptance_on_AWS_CentOS_7_with_EBS
+      timeout: 120
```
Comment (myechuri) [inline build.yaml]: Should this be set to 30 minutes?

File: build.yaml [hunk 5]
```diff
@@ -1600,6 +1600,32 @@ job_type:
...
+                   *setup_aws_env_vars,
+                   'export ACCEPTANCE_YAML=/tmp/acceptance.yml',
+                   *run_trial_with_coverage,
+                   *run_coverage,
+                   *convert_results_to_junit ]
+        }
+      clean_repo: true
+      archive_artifacts: *flocker_artifacts
+      publish_test_results: true
+      # Reasoning as for run_acceptance_on_AWS_CentOS_7_with_EBS
```
Comment (myechuri) [inline build.yaml]: What does this comment mean in the context of running test_installer? Thanks!

File: flocker/acceptance/testtools.py [hunk 6]
```diff
@@ -767,7 +770,7 @@ def get_flocker_version():
 
         d = get_flocker_version()
 
-        # If we fail to get the current version, assume we must reinstall
+        # If we fail to get the current version, assume we must ereinstall
```
Comment (myechuri) [inline flocker/acceptance/testtools.py]: Typo: _ereinstall_

**Comment by myechuri** [comment]
Looks good. Please go ahead once Jenkins job completes, and timeout of 120 in build.yaml is updated. Thanks, @wallrj !

**Comment by wallrj** [comment]
Hmm. Our Jenkins AWS user doesn't have sufficient permissions to create CloudFormation stacks

```
21:30:52 A client error (AccessDenied) occurred when calling the CreateStack operation: User: arn:aws:iam::121466501720:user/vagrant-buildbot is not authorized to perform: cloudformation:CreateStack on resource: arn:aws:cloudformation:us-west-1:121466501720:stack/testinstallerstack1455398911/*
```

**Pull request merged**